src/chemistry_functionality.jl

```diff
@@ -86,7 +86,7 @@ function make_compound(expr)
     # Cannot extract directly using e.g. "getfield.(composition, :reactant)" because then 
     # we get something like :([:C, :O]), rather than :([C, O]).
     composition = Catalyst.recursive_find_reactants!(expr.args[3], 1,
-        Vector{ReactantStruct}(undef, 0))
+        Vector{ReactantInternal}(undef, 0))
```

Comment:
`ReactantInternal` and `ReactionInternal` felt like way better name. Especially now with the `Reaction` struct, having a struct called `ReactionStruct` seemed like it could confuse new users. Since it is used internally only (during the creation of models in the DSL), I figured the name could reflect that.